ls1intum/Artemis
# New model type: Flowchart

### Checklist

### Motivation and Context
This is a new diagram type that is used by several courses.
Implementation is based on https://en.wikipedia.org/wiki/Flowchart

### Steps for Testing

!!!IMPORTANT: After testing the pr, delete the flowchart exercises. Otherwise you cannot test anything else because of the object not found error or transalation not found.

1. Login as instructor
2. Create a modeling exercise with flowchart
3. Create an exam with the that modeling exercise
4. Create another modeling exercise with flowchart as team exercise
5. Login as student
6. Solve modeling exercise as student
7. Attend exam and solve the exercise as that
8. Solve team modeling exercise and try collaboration functionality
9. Login as tutor
10. Assess the modeling exercises
11. Login as student look at your result

Additionally, look for modeling related parts of Artemis to check if anything is missing or broken.

### Test Coverage

Flowchart | 100% (1/1) | 100% (9/9) | 96% (26/27)
FlowchartDecision | 100% (1/1) | 66% (4/6) | 88% (15/17)
FlowchartFlowline | 100% (1/1) | 50% (4/8) | 70% (19/27)
FlowchartFunctionCall | 100% (1/1) | 66% (4/6) | 88% (15/17)
FlowchartInputOutput | 100% (1/1) | 66% (4/6) | 88% (15/17)
FlowchartProcess | 100% (1/1) | 66% (4/6) | 88% (15/17)
FlowchartTerminal | 100% (1/1) | 66% (4/6) | 88% (15/17)
UMLModelParser | 100% (2/2) | 100% (51/51) | 97% (480/493)

### Screenshots

#### Creating flowchart exercise
![out-28](https://user-images.githubusercontent.com/16825065/106010201-78647d80-60b9-11eb-9f1d-c49a9b0be7bd.gif)

#### Solving flowchart exercise
![out-29](https://user-images.githubusercontent.com/16825065/106010216-7bf80480-60b9-11eb-9855-ec31a883c3e5.gif)

#### Assessing flowchart exercise
![out-30](https://user-images.githubusercontent.com/16825065/106010225-7e5a5e80-60b9-11eb-87b1-a26bcfe656af.gif)

#### Viewing assessment result
![out-31](https://user-images.githubusercontent.com/16825065/106010233-80bcb880-60b9-11eb-8a77-4ce6bb5a09ac.gif)
